waves/wallet/src/wallet.rs

```diff
@@ -210,6 +221,258 @@ pub async fn current<'n, 'w>(
     Ok(MutexGuard::map(guard, |w| w.as_mut().unwrap_throw()))
 }
 
+pub async fn withdraw_everything_to(
+    name: String,
+    current_wallet: &Mutex<Option<Wallet>>,
+    address: Address,
+) -> Result<Txid, JsValue> {
+    if !address.is_blinded() {
+        return Err(JsValue::from_str("can only withdraw to blinded addresses"));
+    }
+
+    let wallet = current(&name, current_wallet).await?;
+    let blinding_key = wallet.blinding_key();
+
+    let utxos = map_err_from_anyhow!(esplora::fetch_utxos(&address).await)?;
+    let utxos = utxos
+        .into_iter()
+        .map(|utxo| async move {
+            let mut tx = esplora::fetch_transaction(utxo.txid).await?;
+
+            let txout = tx.output.remove(utxo.vout as usize);
+
+            Result::<_, anyhow::Error>::Ok(match txout {
+                TxOut::Confidential(confidential) => {
+                    let unblinded_txout = confidential.unblind(&*SECP, blinding_key)?;
+
+                    Some((utxo, confidential, unblinded_txout))
+                }
+                TxOut::Explicit(_) => {
+                    log::warn!("spending explicit txouts is unsupported");
+                    None
+                }
+                TxOut::Null(_) => None,
+            })
+        })
+        .collect::<FuturesUnordered<_>>()
+        .filter_map(|result| async { result.transpose() })
+        .try_collect::<Vec<_>>()
+        .await;
+
+    let txouts = map_err_from_anyhow!(utxos)?;
+
+    let prevout_values = txouts
+        .iter()
+        .map(|(utxo, _, unblinded)| {
+            (
+                OutPoint {
+                    txid: utxo.txid,
+                    vout: utxo.vout,
+                },
+                unblinded.value,
+            )
+        })
+        .collect::<HashMap<_, _>>();
+
+    let fee_estimates = map_err_from_anyhow!(esplora::get_fee_estimates().await)?;
+
+    let estimated_virtual_size =
+        estimate_virtual_transaction_size(prevout_values.len() as u64, txouts.len() as u64);
+
+    let fee = (estimated_virtual_size as f64 * fee_estimates.b_6) as u64; // try to get into the next 6 blocks
+
+    let txouts_grouped_by_asset = txouts
+        .into_iter()
+        .group_by(|(_, _, unblinded)| unblinded.asset);
+
+    // prepare the data exactly as we need it to create the transaction
+    let txouts_grouped_by_asset = (&txouts_grouped_by_asset)
+        .into_iter()
+        .map(|(asset, txouts)| {
+            let txouts = txouts.collect::<Vec<_>>();
+
+            // calculate the total amount we want to spend for this asset
+            // if this is the native asset, subtract the fee
```

Comment:
This reminds me that a user can't withdraw without sufficient L-BTC.

Reply:
Yes, for now the UI could just call this function and display the error message? That would say that you need L-BTC.

We can also add an additional check in the UI!

Reply:
I see that the error is handled below.